src/@utils/subgraph.ts

```diff
@@ -223,10 +244,15 @@ export async function fetchData(
 ): Promise<any> {
   try {
     const client = getUrqlClientInstance()
+    const checkUrl = await verifyUrl(context.url)
```

Comment:
not sure about this although it's a nice simple trick. Just concerned that this doubles the requests we make.

When `client.query` does a request, and that errors, that is essentially a URL check so might be worth rather looking into what exactly to react to from the urql response. They have [whole page on error handling](https://formidable.com/open-source/urql/docs/basics/errors/) and somehow they will return a bunch of errors when encountered: https://formidable.com/open-source/urql/docs/api/core/#combinederror. Which should be part of the `OperationResult` we are receiving: https://formidable.com/open-source/urql/docs/api/core/#operationresult

So this might be a more reliable way of detecting subgraph errors. There are also ways to add more what they call [Exchanges](https://formidable.com/open-source/urql/docs/architecture/#the-exchanges), one of them is [errorExchange](https://formidable.com/open-source/urql/docs/api/core/#errorexchange).

That all might take longer to check out and figure out, so if we're in a hurry currently seems to work, but it comes with a heavy price of double the requests.

Reply:
nice, fixed with [76db8b9](https://github.com/oceanprotocol/market/pull/1139/commits/76db8b9464dfd6fa31f82308e68fc41c87b44dce) while I was writing this